### PR TITLE
Add Fabric support

### PR DESCRIPTION
**Pull request by okwasniewski**
Summary:
---------

Introduce Fabric support for iOS and Android. https://github.com/callstack/react-native-slider/issues/380

iOS Changes:

- Setup codegen spec in flow
- Move requireNativeComponent to separate file
- Update podspec file
- Add codegenConfig in package.json
- Ensure backwards compatibility
- iOS Implementation

Android Changes: 

- Configure gradle files
- Implement / extend the generated native interfaces
- Ensure backwards compatibility by splitting implementations into two folders.

Related PRs: 
- https://github.com/callstack/react-native-slider/pull/400
- https://github.com/callstack/react-native-slider/pull/402

Test Plan:
----------

Currently updating example app to RN 0.70 is blocked, tracked here: https://github.com/callstack/react-native-slider/pull/408

###  Test plan for RN <0.70:
 
<details>
<summary>Android</summary>

1. Set `newArchEnabled` to true inside `example/android/gradle.properties`
2. Add this line in `example/android/app/src/main/jni/Android.mk` file.
```diff
+ include $(NODE_MODULES_DIR)/@react-native-community/slider/android/build/generated/source/codegen/jni/Android.mk
include $(CLEAR_VARS)
```
In the same file above, go to the `LOCAL_SHARED_LIBRARIES` setting and add the following line:
```diff
    libreact_codegen_rncore \
+   libreact_codegen_ReactSlider \
    libreact_debug \
```
3. Update `example/android/app/src/main/jni/MainComponentsRegistry.cpp`
```diff
#include <react/renderer/components/rncore/ComponentDescriptors.h>
+ #include <react/renderer/components/ReactSlider/ComponentDescriptors.h>

...

MainComponentsRegistry::sharedProviderRegistry() {
  auto providerRegistry = CoreComponentsRegistry::sharedProviderRegistry();

  // Custom Fabric Components go here. You can register custom
  // components coming from your App or from 3rd party libraries here.
  //
  // providerRegistry->add(concreteComponentDescriptorProvider<
  //        AocViewerComponentDescriptor>());
+ providerRegistry->add(concreteComponentDescriptorProvider<RNCSliderComponentDescriptor>());

  return providerRegistry;
}
```

Then, to launch the project run:
```
npm run install && npm run example-android
```

</details>

<details>
<summary>iOS</summary>

Install pods using a flag
```
RCT_NEW_ARCH_ENABLED=1 pod install
```
Run the app
```
npm run example-ios
```

</details>

### CODE REVIEW

**Review by thymikee — APPROVED** [review]
:shipit:

**Review by BartoszKlonowski — APPROVED** [review]

File: example/ios/Podfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ target 'example' do
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable the next line.
-  use_flipper!()
```
Comment (michalchudziak) [inline example/ios/Podfile:31]: Why this change?

**Review by michalchudziak — APPROVED** [review]
Good job!

File: example/ios/Podfile (hunk 1 above)
Comment (okwasniewski) [reply example/ios/Podfile:31]: Because it significantly sped up the CI, and it's not used that often (if any). Am I right @BartoszKlonowski ?

**Review by deorst — APPROVED** [review]
